### PR TITLE
feat(graphql-http-transformer): support ${region} template in @http url

### LINKED ISSUE #6520: Referencing the Region in the Graphql Schema
**Which Category is your question related to?**
API

**Amplify CLI Version** 
4.39.0

**What AWS Services are you utilizing?**
API, Auth, Function

**Provide additional details e.g. code snippets**
I'm trying to configure an HTTP resolver to call a step function. The url needs the correct aws region. I know you can pass the environment like so `${env}` but I couldn't find anything in the docs stating how to pass the aws region. Is this possible right now or does it need to be feature request?

Something like this: `@http(method: POST, url: "https://states.${region}.amazonaws.com/")`

### PR DESCRIPTION
**Pull request by johnpc**
#### Description of changes

This commit adds support for a `${region}` template, that is replaced with the AWS region of the
environment. It works similarly to the exiting `${env}` template which returns the name of the current
amplify environment.

#### Issue #, if available

fix #6520

re https://github.com/aws-amplify/docs/pull/3203

#### Description of how you validated changes

I added unit tests. I also created an app using the feature it generated the expected data source  in AppSync.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts
Comment (cjihrig) [inline packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts:308]: Please remove this.

File: packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts
Comment (johnpc) [reply packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts]: should this logic be preferred? That is, should the change also be applied to `packages/graphql-http-transformer/src/resources.ts`?

File: packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts
Comment (johnpc) [reply packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts]: good call - I've also done the same rename in `packages/graphql-http-transformer/src/resources.ts`

File: packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts
Comment (johnpc) [reply packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts]: done

File: packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts
Comment (johnpc) [reply packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts]: how embarrassing! consider it gone.

File: packages/amplify-graphql-http-transformer/src/__tests__/amplify-graphql-http-transformer.test.ts
Comment (johnpc) [reply packages/amplify-graphql-http-transformer/src/__tests__/amplify-graphql-http-transformer.test.ts]: correct, that was the intention. I've updated it to reflect that now.

File: packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts
Comment (johnpc) [reply packages/amplify-graphql-http-transformer/src/graphql-http-transformer.ts]: done

**Review by cjihrig — CHANGES_REQUESTED** [review]

**Comment by cjihrig** [comment]
This will require an internal API review before we can merge this.

@renebrandel this PR also updates v1 of the GraphQL Transformer. Do you want that?

**Comment by renebrandel** [comment]
I think this feature makes sense. Let's get a couple of 👍 internally as well. @cjihrig can you port this also to v2 transformer?

**Comment by cjihrig** [comment]
@renebrandel this PR covers v2 as well 👍

**Comment by renebrandel** [comment]
awesome! LGTM from my end.

File: packages/amplify-graphql-http-transformer/src/__tests__/amplify-graphql-http-transformer.test.ts
Comment (johnpc) [reply packages/amplify-graphql-http-transformer/src/__tests__/amplify-graphql-http-transformer.test.ts]: done.

**Review by cjihrig — APPROVED** [review]
LGTM pending API review.

**Comment by johnpc** [comment]
Had an API Review meeting with @edwardfoyle @yuth and Praveen. Changing the template from `${region}` to `${aws_region}`.

**Pull request merged**
